ext/openssl/openssl.c

```diff
@@ -857,8 +864,8 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 			to_add_len = ASN1_STRING_to_UTF8(&to_add, str);
 			needs_free = 1;
 		} else {
-			/* ASN1_STRING_data(3): Since this is an internal pointer it should not be freed or modified in any way */
-			to_add = ASN1_STRING_data(str);
+			/* ASN1_STRING_get0_data(3): Since this is an internal pointer it should not be freed or modified in any way */
+			to_add = (char *)ASN1_STRING_get0_data(str);
```

Comment:
The main reason why `ASN1_STRING_data` is deprecated is due to missing `const` operator for the returned data and thus not enforcing that data won't be modified.

However from the quick look I don't think that you actually need to copy any data because `add_next_index_stringl` already accepts const key so you can define `to_add` as `const char *` which you can safely cast to. The only thing that you will need to do is to address possible warning when calling `ASN1_STRING_to_UTF8` (think there might be warning if you pass const ptr to ptr when non const ptr to ptr expected but not 100% sure). If there is a warning, then it should be very easy to fix though. Maybe something like this will do (not tested):
```
if (ASN1_STRING_type(str) != V_ASN1_UTF8STRING) {
	char *to_add_buf;
	/* ASN1_STRING_to_UTF8(3): The converted data is copied into a newly allocated buffer */
	to_add_len = ASN1_STRING_to_UTF8(&to_add_buf, str);
	to_add = to_add_buf;
	needs_free = 1;
} 
```